internal/service/ec2/ec2_instance.go

```diff
@@ -395,8 +395,8 @@ func ResourceInstance() *schema.Resource {
 						"market_type": {
 							Type:         schema.TypeString,
 							Optional:     true,
```

Comment:
No it doesn't but the fact that `instance_market_options` and all it's sub-attributes are `Computed` (so as to avoid diffs and resource replacement if the instance uses a launch template that specifies instance_market_options`) means that `instance_market_options {}` in configuration is now ignored. That's why I had to change the new test case to add en explicit `market_type = "spot"`.